ldap2pg/script.py

```diff
@@ -52,13 +52,75 @@ def wrapped_main():
     manager.sync(map_=config['sync_map'])
 
 
+class MultilineFormatter(logging.Formatter):
+    def format(self, record):
+        s = super(MultilineFormatter, self).format(record)
+        if '\n' not in s:
+            return s
+
+        lines = s.splitlines()
+        d = record.__dict__.copy()
+        for i, line in enumerate(lines[1:]):
+            record.message = line
+            lines[1+i] = self._fmt % record.__dict__
+        record.__dict__ = d
+
+        return '\n'.join(lines)
+
+
+class ColorFormatter(MultilineFormatter):
+
+    _color_map = {
+        logging.DEBUG: '37',
+        logging.INFO: '1;39',
+        logging.WARN: '96',
+        logging.ERROR: '91',
+        logging.CRITICAL: '1;91',
+    }
+
+    def format(self, record):
+        lines = super(ColorFormatter, self).format(record)
+        color = self._color_map.get(record.levelno, '39')
+        lines = ''.join([
+            '\033[0;%sm%s\033[0m' % (color, line)
+            for line in lines.splitlines(True)
+        ])
+        return lines
+
+
+def logging_dict(tty=True, debug=False):
+    formatter_kwargs = {'class': __name__ + '.ColorFormatter'} if tty else {}
+    return {
+        'version': 1,
+        'formatters': {
+            'debug': dict(
+                format='[%(name)-16s %(levelname)8s] %(message)s',
+                **formatter_kwargs
+            ),
+            'info': dict(format='%(message)s', **formatter_kwargs),
+        },
+        'handlers': {'stderr': {
+            '()': 'logging.StreamHandler',
+            'formatter': 'debug' if debug else 'info',
+        }},
+        'root': {
+            'level': 'WARNING',
+            'handlers': ['stderr'],
+        },
+        'loggers': {
+            'ldap2pg': {
+                'level': 'DEBUG' if debug else 'INFO',
+            },
+        },
+    }
+
+
 def main():
     debug = os.environ.get('DEBUG', '').lower() in {'1', 'y'}
-    logging.basicConfig(
-        level=logging.DEBUG if debug else logging.INFO,
-        format='%(levelname)5.5s %(message)s'
-    )
+    logging_config = logging_dict(debug=debug, tty=sys.stderr.isatty())
+    logging.config.dictConfig(logging_config)
```

Comment:
@julmon @pgiraud with dict config, its easier to test logging config without touching global logging setup.